### PR TITLE
samples: cellular: Updated samples with modem lib shutdown()

### PR DESCRIPTION
**Pull request by MechanicV1**
This PR adds nrf_modem_lib_shutdown() to gracefully shutdown modem on completion of the samples

### CODE REVIEW

File: samples/cellular/battery/src/main.c [hunk 1]
```diff
@@ -424,6 +424,11 @@ int main(void)
 
 		k_sleep(K_SECONDS(refresh_interval));
 	}
+	
+	err = nrf_modem_lib_shutdown();
```
Comment (eivindj-nordic) [inline samples/cellular/battery/src/main.c:428]: This is unreachable per now. The return statements (on error) in above while loop should shut down the modem library. Maybe add a `clean_up` goto as in the ciphersuites sample?

File: samples/cellular/coap_client/src/main.c [hunk 2]
```diff
@@ -185,6 +185,18 @@ static void modem_configure(void)
 #endif /* defined(CONFIG_LTE_LINK_CONTROL) */
 }
 
+/**@brief Truns of LTE link and Shuts down modem to provide LTE link.
+ */
+static void modem_shutdown(void)
+{
+#if defined(CONFIG_LTE_LINK_CONTROL)
+	lte_lc_power_off();	
+#endif /* defined(CONFIG_LTE_LINK_CONTROL) */
```
Comment (eivindj-nordic) [inline samples/cellular/coap_client/src/main.c:194]: We should do a CFUN=0 here if the LTE LC is not enabled.

File: samples/cellular/pdn/src/main.c [hunk 3]
```diff
@@ -163,6 +163,11 @@ int main(void)
 	ifaddrs_print();
 
 	lte_lc_power_off();
+	
+	err = nrf_modem_lib_shutdown();
+	if (err) {
+		printk("Failed to shutdown modem lib, err %d\n", err);
+	}
```
Comment (eivindj-nordic) [inline samples/cellular/pdn/src/main.c:170]: nit: space after end of if bracket.

File: samples/cellular/coap_client/src/main.c (hunk 2 above)
Comment (MechanicV1) [reply samples/cellular/coap_client/src/main.c:194]: nrf_modem_lib_shutdown() this does a cfun 0 after getting the status of device.

File: samples/cellular/coap_client/src/main.c [hunk 4]
```diff
@@ -185,6 +185,19 @@ static void modem_configure(void)
 #endif /* defined(CONFIG_LTE_LINK_CONTROL) */
 }
 
+/**@brief Truns of LTE link and Shuts down modem to provide LTE link.
+ */
+static void modem_shutdown(void)
+{
+#if defined(CONFIG_LTE_LINK_CONTROL)
```
[the remark this answers is not in the document]
Comment (lemrey) [reply samples/cellular/coap_client/src/main.c:192]: Same thing for `modem_configure()` where again the call to `lte_lc_init_and_connect()` is `#ifdef`d.

**Pull request closed without merging**